src/components/Insights/ActiveFiltersView.js

```diff
@@ -5,24 +5,40 @@ import FontIcon from 'material-ui/FontIcon';
 import '../../styles/Insights/ActiveFiltersView.css';
 import { DEFAULT_EXTERNAL_SOURCE, DEFAULT_DATA_SOURCE } from '../../actions/constants';
 
+function areSetsEqual(s1, s2) {
+  if (s1 == null && s2 == null) return true;
+  if (s1 == null || s2 == null) return false;
+  if (s1.size !== s2.size) return false;
+
+  let allEqual = true;
+  s1.forEach(item => {
+    if (!s2.has(s1)) {
+      allEqual = false;
+    }
+  });
+  return allEqual;
+}
+
 class ActiveFiltersView extends React.Component {
 
   shouldComponentUpdate(nextProps, nextState) {
-    const nextplaceid = nextProps.selectedplace.placeid || "";
-    const prevplaceid = this.props.selectedplace.placeid || "";
+    const nextplaceid = (nextProps.selectedplace && nextProps.selectedplace.placeid) || "";
+    const prevplaceid = (this.props.selectedplace && this.props.selectedplace.placeid) || "";
 
-    if(nextProps && nextplaceid === prevplaceid && 
-       nextProps.externalsourceid === this.props.externalsourceid && 
-       nextProps.dataSource === this.props.dataSource) {
+    if (nextProps && nextplaceid === prevplaceid &&
+        nextProps.externalsourceid === this.props.externalsourceid &&
+        nextProps.maintopic === this.props.maintopic &&
+        areSetsEqual(nextProps.termFilters, this.props.termFilters) &&
+        nextProps.dataSource === this.props.dataSource) {
       return false;
     }
 
     return true;
   }
 
   getChips = () => {
-    let chips = [];
-    const { dataSource, externalsourceid, selectedplace } = this.props;
+    const { dataSource, externalsourceid, selectedplace, maintopic, termFilters } = this.props;
+    const chips = [];
```

Comment:
why is `chips` defined as a `const` if it's mutable?

Reply:
It's a mutable object, not a mutable reference. Same reason why this works in Java:

```java
final List<String> foo = new ArrayList<>();

// this works since foo is a mutable type
foo.add("something");

// this doesn't work since foo is an immutable reference
foo = new ArrayList<>();
```